test/apps/test_bayesian_ar_detect_app.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+if [[ $# -ne 14 ]]
+then
+    echo "usage: test_bayesian_ar_detect_app.sh" \
+         "[app_exec] [input_regex_option] [input_regex] " \
+         "[f_time_template_option] [f_time_template]" \
+         "[t_axis_var_option] [t_axis_var]" \
+         "[t_units_option] [t_axis_var]" \
+         "[t_units_option] [t_units]" \
+         "[output_file_option] [output_file]" \
+         "[verifier_exec] [output_file_type] [output_ref]"
+    exit 1
+fi
+
+# First command
+app_exec=${1}
+input_regex_option=${2}
+input_regex=${3}
+f_time_template_option=${4}
+f_time_template=${5}
+t_axis_var_option=${6}
+t_axis_var=${7}
+t_units_option=${8}
+t_units="${9}"
+output_file_option=${10}
+output_file="${11}"
+
+# Second command
+verifier_exec=${12}
+output_file_type=${13}
+output_ref="${14}"
+
+eval ${app_exec} ${input_regex_option} ${input_regex} \
```

Comment:
I think they are necessary with space-separated arguments (like --t_units) in cmake. I only used it for TECA-BARD & Deeplabv3p apps. I tried multiple ways to run the commands and only `eval` was able to run the command successfully.

Reply:
what you are describing is a quoting issue.

Reply:
cmake_parse_arguments (used by teca_add_test) skips "". this is an unfortunate feature of cmake. one can avoid this by calling add_test directly (eg see Python connected components test). Other tests use an optional command line argument set to the empty string by default (eg see CF reader test). There are a number of other easy solutions to avoid use of eval. 374ab79 is a simple solution since it avoids the issue altogether.